app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -381,44 +343,36 @@ class ThreadFragment : Fragment() {
         }
     }
 
-    private fun observeSubjectUpdateTriggers() {
+    private fun observeSubjectUpdateTriggers() = with(binding) {
         threadViewModel.assembleSubjectData(mainViewModel.mergedContactsLive).observe(viewLifecycleOwner) { result ->
-            result.thread?.let {
-                setSubject(
-                    thread = it,
-                    emailDictionary = result.mergedContacts ?: emptyMap(),
-                    aliases = result.mailbox?.aliases ?: emptyList(),
-                    externalMailFlagEnabled = result.mailbox?.externalMailFlagEnabled ?: false,
-                )
-            }
-        }
-    }
 
-    private fun setSubject(
-        thread: Thread,
-        emailDictionary: MergedContactDictionary,
-        aliases: List<String>,
-        externalMailFlagEnabled: Boolean,
-    ) = with(binding) {
-        val (subject, spannedSubject) = computeSubject(thread, emailDictionary, aliases, externalMailFlagEnabled)
-        threadSubject.text = spannedSubject
-        toolbarSubject.text = subject
-
-        threadSubject.setOnLongClickListener {
-            context.copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
-            true
-        }
-        toolbarSubject.setOnLongClickListener {
-            context.copyStringToClipboard(subject, R.string.snackbarSubjectCopiedToClipboard, mainViewModel.snackBarManager)
-            true
+            val (subject, spannedSubject) = computeSubject(
+                thread = result.thread ?: return@observe,
+                emailDictionary = result.mergedContacts ?: emptyMap(),
+                aliases = result.mailbox?.aliases ?: emptyList(),
+                externalMailFlagEnabled = result.mailbox?.externalMailFlagEnabled ?: false,
+            )
```

Comment:
This also need to be in the `ViewModel`.

Reply:
Same answer as https://github.com/Infomaniak/android-kMail/pull/1572?w=1#discussion_r1420239922